docs/tuning-guide.md

```diff
@@ -294,3 +294,34 @@ performance.
 - [`spark.rapids.sql.hasNans`](configs.md#sql.hasNans)
 - [`spark.rapids.sql.castFloatToString.enabled`](configs.md#sql.castFloatToString.enabled)
 - [`spark.rapids.sql.castStringToFloat.enabled`](configs.md#sql.castStringToFloat.enabled)
+
+## Metrics
+
+Custom Spark SQL Metrics are available which can help identify performance bottlenecks in a query.
+
+| Key              | Name                     | Description                                       |
+|------------------|--------------------------|---------------------------------------------------|
+| bufferTime       | buffer time              | Time spent buffering input from file data sources. |
+| buildDataSize    | build side size          | Size in bytes of the build-side of a join.        |
+| buildTime        | build time               | Time to load the build-side of a join.            |
+| collectTime      | collect time             | Time spent collecting data from child operator(s).|
+| computeAggTime   | aggregation time         | Time performing aggregation.                      |
+| concatTime       | concat batch time        | Time to concatenate batches.                      |
+| filterTime       | filter time              | Time spent applying filters within other operators, such as joins. |
+| gpuDecodeTime    | GPU decode time          | Time spent on GPU decoding encrypted or compressed data. |
+| joinOutputRows   | join output rows         | The number of rows produced by a join before any filter expression is applied. |
+| joinTime         | join time                | Total time for performing a join.                 |
+| numInputBatches  | input columnar batches   | Number of columnar batches that the operator received from its child operator(s). |
```

Comment:
Some of these metrics, like this one, will not appear by default.  Does this table need an extra column explaining the GpuMetric levels and how the user can enable the metrics?  Eventually I think we'd want to automatically generate this documentation from the code, because otherwise it's destined to be constantly out of date.  Not something for this PR, but we may want an issue to track.

Reply:
Agreed. I had already filed https://github.com/NVIDIA/spark-rapids/issues/2510 for generating this documentation as part of the build. I'll add some information about the levels.